vernac/g_vernac.mlg

```diff
@@ -442,22 +442,22 @@ GRAMMAR EXTEND Gram
   ;
   record_binder:
     [ [ id = name -> { (NoInstance,AssumExpr(id, [], CAst.make ~loc @@ CHole (None, IntroAnonymous, None))) }
-      | id = name; f = record_binder_body -> { f id } ] ]
+      | id = name; f = field_body -> { f id } ] ]
   ;
```

Comment:
> The entry `record_binder` may become clearer if one inlines `record_binder_body` into `record_binder`.

While possible, we also considered layout criteria in coming up with the documented grammar.  We prefer to keep each production short enough so that it fits on a single line for the sake of readability.  I think inlining `field_body` would be harder to read.  (BTW, if it makes sense for `field_body` to just be `{* binder }`, that would let `field_body` have a single line definition with `of_type` and `:= term` as optional items.  But not in this PR.)

Here's the documentation for `Record`, one of the most complicated-looking parts of the grammar.  (Later on, I'd like to add some concrete examples of records to make it a little easier to digest.)

![image](https://user-images.githubusercontent.com/1253341/96642900-a9bff700-12db-11eb-844b-2ec632dac332.png)

> The name `record_binder` could itself be `record_field_decl`

Sure.  `record_binder` is spliced away in the documentation.

Reply:
If we want the syntax to reflect the semantics, there are only two kinds of record fields, assumptions and definitions:
```
record_field ::= attribute* name (field_assum | field_def) priority notations
field_assum  ::= binder* of_type?
field_def    ::= binder* of_type? := term
```
which could be generalized into:
```
record_field ::= attribute* name binder* (field_assum | field_def) priority notations
field_assum  ::= of_type?
field_def    ::= of_type? := term
```
and which could also be presented as:
```
record_field ::= attribute* (field_assum | field_def) priority notations
field_assum  ::= name binder* of_type?
field_def    ::= name binder* of_type? := term
```
or
```
record_field ::= attribute* field_decl priority notations
field_decl   ::= field_assum
              |  field_def
field_assum  ::= name binder* of_type?
field_def    ::= name binder* of_type? := term
```
etc.

Reply:
In your comment, you made the `of_type` of `field_assum` be optional, but it doesn't seem it is in the documentation shown by Jim here.

Reply:
There is a "?" on the `field_body` box, which means that the name can be given alone (implicitly standing for `id : _`).

Then, I proposed the generalization to accept the name coming with binders but still no type (standing for `id binders : _`)

Reply:
I don't fully follow your suggestions.

> There is a "?" on the `field_body` box, which means

Which means that the `field_body` is optional.  It doesn't make `{* binder }` alone a valid `field_assum`.  If we want to allow that, which AFAICT is reasonable, it seems to me that it should be a different PR--it doesn't fit so well with the renaming focus of this PR.

> there are only two kinds of record fields, assumptions and definitions

Splitting `field_body` into `field_assum` and `field_def` may make sense, but note that there is currently zero explanation of `field_body` in the doc.  Not sure there's much point in splitting this up now without adding explanation in the doc, which I wouldn't want to do in this PR.  There's still lots of work to do on the doc to make sure that every bit of syntax is explained and explained well.

Also unclear whether you mean the mlg should make this distinction or just the doc.  We'd like to keep the doc grammar close to the mlg grammar with occasional exceptions.  The text we eventually provide for the current `field_body` or its replacement could clearly distinguish between the assumption and declaration cases even if the grammar is unchanged.  That's probably sufficient for readers.

Reply:
> If we want to allow that, which AFAICT is reasonable, it seems to me that it should be a different PR--it doesn't fit so well with the renaming focus of this PR.

Yes, it can be a different PR. So, ok as it is now.

> Also unclear whether you mean the mlg should make this distinction or just the doc.

I meant first the doc (which currently does well reflect the underlying semantics), and possibly also the mlg.